### PR TITLE
Adds CORS headers to recommendation endpoint (closes #36).

### LINKED ISSUE #36: Add CORS headers to responses.

### PR DESCRIPTION
**Pull request by chuckharmston**
*To-do:*

- [x] Add tests

Creating PR to let CI do its thing.

### CODE REVIEW

File: app/cors.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    `max_age` - The number of seconds as integer or timedelta object for which
+        the preflighted request is valid.
+    `attach_to_all` - True if the decorator should add the access control
+        headers to all HTTP methods or False if it should only add them to
+        OPTIONS responses.
+    `automatic_options` - If enabled the decorator will use the default Flas
+        OPTIONS response and attach the headers there, otherwise the view
+        function will be called to generate an appropriate response.
+
+    Adapted from http://flask.pocoo.org/snippets/56/
```
Comment (6a68) [inline app/cors.py]: It looks like https://pypi.python.org/pypi/Flask-Cors/ is based on the same snippet, could we use that instead? Less code to write = win

**Comment by 6a68** [comment]
Just one question, back to you @chuckharmston :-)

File: app/cors.py (hunk 1 above)
Comment (chuckharmston) [inline app/cors.py]: Oh, absolutely. I somehow missed that and had a personal todo to move it to a package with Armin's permission.

File: app/tests/test_main.py [hunk 2]
```diff
@@ -51,8 +64,7 @@ def test_exception(self, mock_get):
         eq_(response.json, {})
 
     @patch('app.tests.memcached.mock_memcached.get')
-    @patch('app.main.recommend.delay')
-    def test_cache_hit(self, mock_delay, mock_get):
+    def test_cache_hit(self, mock_get):
```
Comment (chuckharmston) [inline app/tests/test_main.py]: Drive-by removal of a mock that wasn't being used.

**Comment by chuckharmston** [comment]
Flask-Cors is unfortunately not compatible with our version of Python, but your "less code" point is well-taken: all we really need to do is add three headers to all requests. I stripped out the decorator in the original commit and replaced it with a very simple response middleware to do so.

**Comment by chuckharmston** [comment]
Closing/reopening because CI seems to be stuck.

**Comment by 6a68** [comment]
lgtm! merging
